pkg/workloads/console/controller.go

```diff
@@ -347,21 +383,30 @@ func (r *reconciler) clampTimeout(template *workloadsv1alpha1.ConsoleTemplate) e
 }
 
 func (r *reconciler) updateStatus(job *batchv1.Job) error {
-	// Fetch the job's pod
-	podList := &corev1.PodList{}
-	pod := &corev1.Pod{}
-	opts := client.
-		InNamespace(r.name.Namespace).
-		MatchingLabels(map[string]string{"job-name": job.ObjectMeta.Name})
-	if err := r.client.List(r.ctx, opts, podList); err != nil {
-		return err
-	}
-	if len(podList.Items) > 0 {
-		pod = &podList.Items[0]
+	var newStatus workloadsv1alpha1.ConsoleStatus
+
+	if job == nil {
```

Comment:
Putting a conditional here seems a bit wrong. Can we not instead conditionally wrap the bit of the code that does the pod listing so that it doesn't attempt to run if the job is nil.

Then we can keep doing all of the status calculation stuff in the `calculateStatus` func?

Reply:
Yeah after looking at the changes in the `calculateStatus` func below, I do think it'd be nicer to keep this logic in the `calculatePhase` function

Reply:
Agreed about this. It was wrong to set a phase outside of `calculatePhase`. I'll rejig this so we do it in there.